backend/api/models.py

```diff
@@ -1,3 +1,174 @@
+        if voted_before:
+            vote_value = vote_value * 2
+        self.vote_count = self.vote_count + vote_value
+        self.save()
+
+
+class GenericModelMixin(models.Model):
+    """
+    Allows generic relations between different models.
+    See https://docs.djangoproject.com/en/2.1/ref/contrib/contenttypes/
+    """
+    content_type = models.ForeignKey(ContentType, on_delete=models.CASCADE)
+    object_id = models.PositiveIntegerField()
+    content_object = GenericForeignKey('content_type', 'object_id')
+
+    class Meta:
+        abstract = True
+
+
+class Event(OwnerMixin, CommentMixin, FollowMixin, LocationMixin, MediaMixin, VoteMixin):
+    # Related fields
+    # TODO Decide if an artist must be a User in our system.
+    artists = models.ManyToManyField(settings.AUTH_USER_MODEL, db_table='event_artists',
+                                     related_name='performed_events', blank=True)
+    tags = models.ManyToManyField('Tag', db_table='event_tags', related_name='events',
+                                  blank=True)
+
+    # Own fields
+    title = models.CharField(max_length=100)
+    description = models.TextField()
+    date = models.DateTimeField(default=timezone.now)
+    price = models.DecimalField(max_digits=6, decimal_places=2, blank=True, default=0.0)
+    organizer_url = models.URLField(blank=True, null=True)
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+
+
+class AttendanceStatus(OwnerMixin):
+    ATTENDANCE_STATUS = (
+        ('Y', 'Yes'),
+        ('N', 'No'),
+        ('M', 'Maybe'),
+        ('A', 'Attended'),
+        ('B', 'Blocked'),
+    )
+    event = models.ForeignKey(Event, related_name='attendance_status', on_delete=models.CASCADE)
+    status = models.CharField(max_length=1, choices=ATTENDANCE_STATUS)
+
+    class Meta:
+        # There can be only one attendance status between User and Event.
+        unique_together = ('owner', 'event')
+
+
+class Comment(GenericModelMixin, OwnerMixin):
+    content = models.TextField()
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+
+
+class FollowStatus(GenericModelMixin, OwnerMixin):
+    class Meta:
+        # A user cannot follow the same item more than once
+        unique_together = ('owner', 'content_type', 'object_id')
+
+
+class Location(GenericModelMixin, OwnerMixin):
+    # TODO Add required fields after doing research about Google Maps / Places API
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+
+    class Meta:
+        # A model (User, Event etc.) cannot have more than one Location.
+        unique_together = ('content_type', 'object_id')
+
+
+class Media(GenericModelMixin, OwnerMixin):
+    url = models.URLField()
+    created = models.DateTimeField(auto_now_add=True)
+    updated = models.DateTimeField(auto_now=True)
+
+
+class Tag(models.Model):
```

Comment:
Maybe we should create a mixin for this too.

Reply:
I'll create a new PR later on.